bundles/org.openhab.binding.publictransportswitzerland/src/main/java/org/openhab/binding/publictransportswitzerland/internal/stationboard/PublicTransportSwitzerlandStationboardHandler.java

```diff
@@ -0,0 +1,252 @@
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        } else {
+            updateStatus(ThingStatus.UNKNOWN);
+            startChannelUpdate();
+        }
+    }
+
+    @Override
+    public void dispose() {
+        stopChannelUpdate();
+    }
+
+    private void startChannelUpdate() {
+        updateChannelsJob = scheduler.scheduleWithFixedDelay(this::updateChannels, 0, 60, TimeUnit.SECONDS);
+    }
+
+    private void stopChannelUpdate() {
+        if (updateChannelsJob != null && !updateChannelsJob.isCancelled()) {
+            updateChannelsJob.cancel(true);
+        }
+    }
+
+    public @Nullable JsonElement updateData() {
+        PublicTransportSwitzerlandStationboardConfiguration config = getConfigAs(PublicTransportSwitzerlandStationboardConfiguration.class);
+
+        try {
+            String escapedStation = URLEncoder.encode(config.station, StandardCharsets.UTF_8.name());
+            String requestUrl = BASE_URL + "stationboard?station=" + escapedStation + FIELD_FILTERS;
+
+            String response = HttpUtil.executeUrl("GET", requestUrl, 10_000);
+            logger.debug("Got response from API: {}", response);
+
+            return new JsonParser().parse(response);
+        } catch (Exception e) {
+            logger.warn("Unable to fetch stationboard data", e);
+            return null;
+        }
+    }
+
+    private static String createFilterForFields(String... fields) {
+        return Arrays.stream(fields).map((field) -> "&fields[]=" + field).collect(Collectors.joining());
+    }
+
+    private void updateChannels() {
+        @Nullable JsonElement jsonObject = cache.getValue();
+
+        if (jsonObject == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+
+            updateState(CHANNEL_TSV, UnDefType.UNDEF);
+
+            for (Channel channel : getThing().getChannelsOfGroup(dynamicChannelGroupUID.getId())) {
+                updateState(channel.getUID(), UnDefType.UNDEF);
+            }
+
+            return;
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+
+        JsonArray stationboard = jsonObject.getAsJsonObject().get("stationboard").getAsJsonArray();
+
+        createDynamicChannels(stationboard.size());
+        setUnusedDynamicChannelsToUndef(stationboard.size());
+
+        List<String> tsvRows = new ArrayList<>();
+
+        for (int i = 0; i < stationboard.size(); i++) {
+            JsonElement jsonElement = stationboard.get(i);
+
+            JsonObject departureObject = jsonElement.getAsJsonObject();
+            JsonObject stopObject = departureObject.get("stop").getAsJsonObject();
+
+            String category = departureObject.get("category").getAsString();
+            String number = departureObject.get("number").getAsString();
+            String destination = departureObject.get("to").getAsString();
+            Long departureTime = stopObject.get("departureTimestamp").getAsLong();
```

Comment:
Better don't rely that these attributes are always present. A missing attribute should not result in a NPE.